File: week2/file_io.py
import time

def main():
    start_time = time.time()  # Start timing

    # Writing to a file
    with open('test_file.txt', 'w') as f:
        for _ in range(10000):  # Writing multiple lines to simulate a large file
            f.write("Hello, world!\n")

    # Reading from the file
    with open('test_file.txt', 'r') as f:
        content = f.read()

    end_time = time.time()  # End timing
    print(f"File I/O took {end_time - start_time:.6f} seconds.")

if __name__ == "__main__":
    main()

File: week2/file_io.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    FILE *fp;
    char buffer[255];

    // Writing to a file
    fp = fopen("example.txt", "w");
    if(fp == NULL) {
        perror("Error opening file");
        return(-1);
    }
    fprintf(fp, "Hello, world!\n");
    fclose(fp);

    // Reading from a file
    fp = fopen("example.txt", "r");
    if(fp == NULL) {
        perror("Error opening file");
        return(-1);
    }
    fgets(buffer, 255, fp);
    printf("File content: %s", buffer);
    fclose(fp);

    return 0;
}

